src/main/scala/com/dwolla/lambda/cloudflare/requests/processors/AccountMembership.scala

```diff
@@ -92,13 +91,11 @@ class AccountMembership[F[_] : Sync](accountsClient: AccountsClient[F], accountM
     }
 
   private def findRequestedRoles(accountId: AccountId, requestedRoleNames: Set[String]): F[List[AccountRole]] =
-    for {
-      roles ← accountsClient.listRoles(accountId).compile.toList
-      () ← Sync[F].delay(logger.debug(s"Requested roles for account: $requestedRoleNames"))
-      () ← Sync[F].delay(logger.debug(s"Found roles for account: $roles"))
-      foundRoles = roles.filter(ar ⇒ requestedRoleNames.contains(ar.name))
-      res ← if (requestedRoleNames == foundRoles.map(_.name).toSet) Applicative[F].pure(roles) else ApplicativeError[F, Throwable].raiseError(MissingRoles(requestedRoleNames.toList))
```

Comment:
This line is the problem (it effectively returns `roles` instead of the filtered list `foundRoles`). The rewrite of this function avoids bringing the entire unfiltered list into scope except for logging purposes, which stylistically should help avoid this class of error.